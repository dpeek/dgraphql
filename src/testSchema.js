/* @flow */

import fs from 'fs'
import path from 'path'

import { buildSchema } from './schema'
import { Client } from './client'
import type { ClientConfig } from './client'

export default async function initSchema (
  graphql: string,
  dgraph: string,
  config: ClientConfig
) {
  const client = new Client(config, {})
  let dgraphPath = path.resolve(__dirname, '__tests__', dgraph)
  await client.fetchQuery(fs.readFileSync(dgraphPath).toString())

  const graphqlPath = path.resolve(__dirname, '__tests__', graphql)
  return buildSchema(fs.readFileSync(graphqlPath).toString(), config)
}
