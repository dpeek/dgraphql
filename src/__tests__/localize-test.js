import fs from 'fs'
import path from 'path'

import { graphql as graphql2 } from 'graphql'
import testSchema from '../testSchema'

let schema

function graphql (schema, source, language) {
  return graphql2({
    schema,
    source,
    contextValue: { language: language }
  })
}

beforeAll(async () => {
  schema = await testSchema('test.graphql', { debug: true })
})

test('queries node field', async () => {
  const create = `mutation {
    createPerson(input: {
      name: "Tim",
      title: "Mister"
    }) {
      person {
        id
      }
    }
  }`
  const createResult = await graphql(schema, create, 'en')
  const id = createResult.data.createPerson.person.id

  const update = `mutation {
    updatePerson(input: {
      id: "${id}",
      name: "Tim",
      title: "Se√±or"
    }) {
      person {
        id
      }
    }
  }`
  await graphql(schema, update, 'es')

  let query = `query {
    person(id: "${id}") {
      name
      title
    }
  }`
  let queryResult = await graphql(schema, query, 'en')
  expect(queryResult).toMatchSnapshot()

  queryResult = await graphql(schema, query, 'es')
  expect(queryResult).toMatchSnapshot()
})
