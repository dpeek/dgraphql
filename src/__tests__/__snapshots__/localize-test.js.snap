// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries node field 1`] = `
Object {
  "data": Object {
    "person": Object {
      "name": "Tim",
      "title": "Mister",
    },
  },
}
`;

exports[`queries node field 2`] = `
Object {
  "data": Object {
    "person": Object {
      "name": "Tim",
      "title": "Se√±or",
    },
  },
}
`;
